azuredevops/internal/service/core/data_project.go

```diff
@@ -17,8 +19,22 @@ func DataProject() *schema.Resource {
 				Type:     v.Type,
 				Computed: true,
 			}
+		} else {
+			v.Required = false
```

Comment:
duplicate with `v.Optional = true`

Reply:
But it's set to be required true, in the original definition

Reply:
I mean `Optional = true` is equal to `Required = false` here

Reply:
Removing required results in:
![image](https://user-images.githubusercontent.com/12761692/91162876-99691280-e6c4-11ea-9e85-0331440daf39.png)

Reply:
You get the `project_name` base schema  from `ResourceProject()`,  the origin schema is:
```
"project_name": {
  Type:             schema.TypeString,
  Required:         true,
  ValidateFunc:     validation.StringIsNotWhiteSpace,
  DiffSuppressFunc: suppress.CaseDifference,
},
```
after you change the schema at code [line 17~21](https://github.com/xuzhang3/terraform-provider-azuredevops/pull/163/files/106b8fa8bf926a2ad109ea89fd81cbba923c62f5#diff-443557ef19497038b4172b9ac825bf0dR17-R21)
```
if k == "project_name" {
  v.Required = false
  v.Optional = true
  baseSchema.Schema[k] = v
}
```
the schema will looks like :
```
"project_name": {
  Type:             schema.TypeString,
  Required:         false,
  Optional:          true,
  ValidateFunc:     validation.StringIsNotWhiteSpace,
  DiffSuppressFunc: suppress.CaseDifference,
},
```
![image](https://user-images.githubusercontent.com/57888764/91249930-6426f800-e78a-11ea-9b3b-34b0b6810711.png)

Suggest rewrite the schema not extend it.

Reply:
Super valid point :) Thanks.
What do you think about my changes?

Reply:
`DiffSuppressFunc: suppress.CaseDifference,` can  be removed.